PIPE2D-852: Added fitBroadbandSED.py

SogoMineo opened the pull request:
FitBroadbandSEDTask is a task to derive a probability distribution
function from broad-band SED fitting of a standard star.
The original program was provided by Takuji Yamashita.

Requires PIPE2D-852 upgrade to drp_pfs_data.

PaulPrice commented on python/pfs/drp/stella/fitBroadbandSED.py:
```diff
@@ -0,0 +1,135 @@
+            "g": "HSCg", "r": "HSCr", "r2": "HSCr2", "i": "HSCi", "i2": "HSCi2",
+            "z": "HSCz", "y": "HSCy"
+        },
+        doc="Conversion table from AFW's filter names to those used by `fluxLibrary`"
```
I don't think we need to worry about AFW's filter names, and there are a lot of changes to AFW's filter handling coming soon so we should avoid it. All we need to worry about are the filter names in the pfsConfig.

PaulPrice commented on python/pfs/drp/stella/fitBroadbandSED.py:
```diff
@@ -0,0 +1,135 @@
+
+        return pdfs
+
+    def run(self, filterNames, fiberFlux, fiberFluxErr):
```
Can we have the `run` method simply take a `pfsConfig`? That makes it easier for users to call directly.

SogoMineo replied:
`pfsConfig` contains multiple spectra, but `run()` here deals with one spectra per call. Should I change `run()` so it will take `pfsConfig`, and rename the old `run()` to `getProbabilities()`?

PaulPrice replied:
Yeah, that sounds good!

PaulPrice commented on python/pfs/drp/stella/fitBroadbandSED.py:
```diff
@@ -0,0 +1,135 @@
+        doc="Synthetic photometry table"
+    )
+
+    filterAliases = DictField(
```
I'd call this `filterMappings`: it's a mapping from the pfsConfig filter name to the flux library column. But not really important.

PaulPrice commented on python/pfs/drp/stella/fitBroadbandSED.py:
```diff
@@ -0,0 +1,135 @@
+        # Convert filter names.
+        filterNames = [self.config.filterAliases.get(f, f) for f in filterNames]
+
+        # fluxLibrary.shape = (nSEDs, nBands)
```
This looks like code, and having commented-out code is a violation of the style guide, so I suggest just prefixing it with "Note:". Ditto below.

PaulPrice commented on python/pfs/drp/stella/fitBroadbandSED.py:
```diff
@@ -0,0 +1,135 @@
+from lsst.pipe.base import Task
+from lsst.utils import getPackageDir
+
+import numpy
```
We usually `import numpy as np`, but it's up to you.

PaulPrice commented on python/pfs/drp/stella/fitBroadbandSED.py:
```diff
@@ -0,0 +1,135 @@
+        pfsConfig = dataRef.get("pfsConfig")
+
+        pdfs = []
+        for filterNames, fiberFlux, fiberFluxErr \
```
Don't you need to select the fluxcal fibers, rather than iterating over all fibers?

PaulPrice reviewed (approved):
I suggest a few changes, but this is great!

SogoMineo commented:
I have changed `run()` method so that it now deals with pfsConfig. Fibers whose `targetType` is not `TargetType.FLUXSTD` are skipped. I also changed the unit test largely.
